browser/ui/webui/new_tab_page/brave_new_tab_message_handler.cc

```diff
@@ -609,6 +627,20 @@ void BraveNewTabMessageHandler::HandleTodayOnCardViews(
                              base::size(kBuckets) + 1);
 }
 
+void BraveNewTabMessageHandler::HandleTodayOnPromotedCardView(
+    const base::ListValue* args) {
+  // Argument should be how many cards viewed in this session.
+  std::string creative_instance_id = args->GetList()[0].GetString();
+  std::string item_id = args->GetList()[1].GetString();
+  if (!item_id.empty() && !creative_instance_id.empty()) {
+    auto* ads_service_ = brave_ads::AdsServiceFactory::GetForProfile(profile_);
+    ads_service_->OnBraveTodayAdEvent(
```

Comment:
Changed to `OnPromotedContentAdEvent` in latest ads library changes

Reply:
Changed